src/auth/auth-config.ts

```diff
@@ -773,6 +792,63 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
         '"OIDCAuthProviderConfig.displayName" must be a valid string.',
       );
     }
+    if (typeof options.clientSecret !== 'undefined' &&
+        !validator.isNonEmptyString(options.clientSecret)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_CONFIG,
+        '"OIDCAuthProviderConfig.clientSecret" must be a valid string.',
+      );
+    }
+    if (typeof options.responseType !== 'undefined') {
+      let idTokenType = false;
+      let codeType = false;
+      for (const responseTypeKey in options.responseType) {
+        if (!(responseTypeKey in validResponseTypes)) {
+          throw new FirebaseAuthError(
+            AuthClientErrorCode.INVALID_CONFIG,
+            `"${responseTypeKey}" is not a valid OAuthResponseType parameter.`,
+          );
+        } else {
+          if (responseTypeKey == 'idToken') {
+            if (!validator.isBoolean(options.responseType.idToken)) {
+              throw new FirebaseAuthError(
+                AuthClientErrorCode.INVALID_ARGUMENT,
+                `"OIDCAuthProviderConfig.responseType.${responseTypeKey}" must be a boolean.`,
+              );
+            }
+            if (options.responseType && options.responseType.idToken) {
+              idTokenType = options.responseType.idToken;
+            }
+          } else if (responseTypeKey == 'code') {
+            if (!validator.isBoolean(options.responseType.code)) {
+              throw new FirebaseAuthError(
+                AuthClientErrorCode.INVALID_ARGUMENT,
+                `"OIDCAuthProviderConfig.responseType.${responseTypeKey}" must be a boolean.`,
+              );
+            }
+            if (options.responseType && options.responseType.code) {
+              codeType = options.responseType.code;
+            }
+          }
+        }
+      }
+
+      // Exact one of OAuth response type needs to be set to true.
+      if ((idTokenType && codeType) ||
+          (!idTokenType && !codeType)) {
```

Comment:
This condition will fire if the developer does something like `options.responseType = {}`. I think that's not technically an error?

Reply:
ok, I added 2 parameters checking whether responseType has been reset.